src/ZODB/tests/testFileStorage.py

```diff
@@ -326,6 +326,35 @@ def checkFlushNeededAfterTruncate(self):
             self._storage._files.flush = lambda: None
             self.checkFlushAfterTruncate(True)
 
+    def checkCommitWithEmptyData(self):
+        """
+        Verify that transaction is persisted even if it has no data, or even
+        both no data and empty metadata.
+        """
+
+        # verify:
+        # - commit with empty data but non-empty metadata
+        # - commit with empty data and empty metadata
+        #   (the fact of commit carries information by itself)
+        stor = self._storage
+        for description in [u'commit with empty data', u'']:
```

Comment:
Can you remove the square brackets ?

Reply:
Do you mean change list -> tuple?

Reply:
Yes. But parentheses aren't required.

Reply:
After being with python for ~20 years I've discovered that `for x in 1,2,3:` (i.e. without brackets) also works. I don't think this is cleanier, but if required I can indeed change to this style.

For the reference [ZODB article](https://github.com/zopefoundation/ZODB/blob/master/doc/articles/ZODB1.rst) uses the same style I used initially:

```
>>> for name in ['Mary', 'Joe', 'Bob']:
        ...
```

Reply:
My main concern is about the fact that lists have to be built at runtime, whereas immutable objects (here a tuple of strings) don't.

Reply:
It indeed used to be so on python2, but on python3 it does not produce any difference as iterating through both constant tuples and lists produces exactly the same bytecode (using tuple inside). At the same time I find the style where iterated list/tuple is explicitly marked as such - using either `[]` or `()` to be more clear. While personally I find `[` to be more visually clear, I've changed the patch to use tuple explicitly as requested. 

```py
def f_tuple():
    for i in ('hello', 'world', 'zzz'):
        pass

def f_list():
    for i in ['hello', 'world', 'zzz']:
        pass

from dis import dis
print('\nTUPLE:')
dis(f_tuple)
print('\nLIST:')
dis(f_list)
```

```
$ python3 x.py 

TUPLE:
  2           0 SETUP_LOOP              12 (to 14)
              2 LOAD_CONST               1 (('hello', 'world', 'zzz'))
              4 GET_ITER
        >>    6 FOR_ITER                 4 (to 12)
              8 STORE_FAST               0 (i)

  3          10 JUMP_ABSOLUTE            6
        >>   12 POP_BLOCK
        >>   14 LOAD_CONST               0 (None)
             16 RETURN_VALUE

LIST:
  6           0 SETUP_LOOP              12 (to 14)
              2 LOAD_CONST               1 (('hello', 'world', 'zzz'))
              4 GET_ITER
        >>    6 FOR_ITER                 4 (to 12)
              8 STORE_FAST               0 (i)

  7          10 JUMP_ABSOLUTE            6
        >>   12 POP_BLOCK
        >>   14 LOAD_CONST               0 (None)
             16 RETURN_VALUE
```

```
kirr@deco:~/tmp/trashme$ python2 x.py 

TUPLE:
  2           0 SETUP_LOOP              14 (to 17)
              3 LOAD_CONST               4 (('hello', 'world', 'zzz'))
              6 GET_ITER            
        >>    7 FOR_ITER                 6 (to 16)
             10 STORE_FAST               0 (i)

  3          13 JUMP_ABSOLUTE            7
        >>   16 POP_BLOCK           
        >>   17 LOAD_CONST               0 (None)
             20 RETURN_VALUE        

LIST:
  6           0 SETUP_LOOP              23 (to 26)
              3 LOAD_CONST               1 ('hello')
              6 LOAD_CONST               2 ('world')
              9 LOAD_CONST               3 ('zzz')
             12 BUILD_LIST               3
             15 GET_ITER            
        >>   16 FOR_ITER                 6 (to 25)
             19 STORE_FAST               0 (i)

  7          22 JUMP_ABSOLUTE           16
        >>   25 POP_BLOCK           
        >>   26 LOAD_CONST               0 (None)
             29 RETURN_VALUE        
```